src/server/utils/bucket_diff.js

```diff
@@ -165,9 +168,12 @@ class BucketDiff {
      * @returns {nb.BucketDiffKeysDiff}
      */
     _object_grouped_by_key_and_omitted(list) {
-        if (!list) return {};
+        if (!list || (this.for_deletion && !this.version)) return {};
         dbg.log1('_object_grouped_by_key_and_omitted list:', list);
-        const field = this.version ? "Versions" : "Contents";
+        let field = this.version ? "Versions" : "Contents";
+        if (this.for_deletion && this.version) {
```

Comment:
Isn't it redundant with `if (!list || (this.for_deletion && !this.version)) return {};` ?
If we are here and` if (this.for_deletion)`, then `this.version` must be true.
```suggestion
        if (this.for_deletion) {
```

Reply:
Sure, will update it.